### PR TITLE
Icebox QoL minus features

### PR DESCRIPTION
**Pull request by Whoneedspacee**
## About The Pull Request

Puts Ice Hiking Boots in the mining vendor for the same credits as jump boots.

Adds GPS Signals for megafauna and icebox spawners.

Increases the demonic frost miners projectile speed slightly and changes up his arena.

Makes it so ice whelp breath temporarily freezes you in place.

Makes it so plasma lava boats can be created from polar bear hide.

Adds cold protection to the ash drake armor.

Adds the ability to mine weak mineral turfs with your hands.

Adds openspace surrounding the icebox gulag and mining base.

Fixes icebox lavaland and asteroid ruins not generating.

Icebox can now rarely spawn the colossus on it's bottom level.

## Why It's Good For The Game

Icebox is buggy and this pr is basically pest repellant.

## Changelog
:cl: Whoneedspacee
add: You can now purchase ice hiking boots from the mining vendor.
add: You can now craft plasma lava boats from polar bear hide.
add: Icebox turfs can now be mined with your bare hands.
tweak: Icebox creatures now have signals that can be tracked.
balance: Ash drake armor now has cold protection so it can be used on icebox.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
can you make wolf NOT BREAK WALLS and other mobs not push windows??

**Comment by ATH1909** [comment]
> Adds the ability to mine weak mineral turfs with your hands.

👀 

does this apply to lavaland's weak turfs as well? how does this interact with improvised pickaxes?

**Review by Twaticus — CHANGES_REQUESTED** [review]
not sure if the mob icon removal is intended but the suit needs to be fixed
image

**Review by Twaticus — APPROVED** [review]

**Comment by Whoneedspacee** [comment]
conflicto resolvo

**Comment by Maurukas** [comment]
The large diff looks like it's just the area changes being counted. Looks good when I check locally.

**Comment by Whoneedspacee** [comment]
save me from map conflicts

**Comment by Mothblocks** [comment]
thats what you get for not atomizing ur stuff

File: code/game/turfs/closed/minerals.dm [hunk 1]
```diff
@@ -80,6 +82,23 @@
 	else
 		return attack_hand(user)
 
+/turf/closed/mineral/attack_hand(mob/user)
+	if(weak_turf)
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:86]: Early return.

File: code/game/turfs/closed/minerals.dm [hunk 2]
```diff
@@ -80,6 +82,23 @@
...
+		var/turf/T = user.loc
+		if (!isturf(T))
+			return
+		if(last_act + 20 > world.time)//prevents message spam
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:90]: Magic number--move to define.

File: code/game/turfs/closed/minerals.dm [hunk 3]
```diff
@@ -80,6 +82,23 @@
 	else
 		return attack_hand(user)
 
+/turf/closed/mineral/attack_hand(mob/user)
+	if(weak_turf)
+		var/turf/T = user.loc
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:87]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/game/turfs/closed/minerals.dm [hunk 4]
```diff
@@ -80,6 +82,23 @@
...
+		if (!isturf(T))
+			return
+		if(last_act + 20 > world.time)//prevents message spam
+			return
+		last_act = world.time
+		to_chat(user, "<span class='notice'>You start pulling out pieces of rock with your hands...</span>")
+
+		if(do_after(user, 15 SECONDS, target = src))
+			if(ismineralturf(src))
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:96]: Combine these into one or early return.

File: code/game/turfs/closed/minerals.dm [hunk 5]
```diff
@@ -80,6 +82,23 @@
...
+				to_chat(user, "<span class='notice'>You finish pulling apart the rock.</span>")
+				gets_drilled(user, FALSE)
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:98]:
```suggestion
				gets_drilled(user)
```

File: code/game/turfs/closed/minerals.dm [hunk 6]
```diff
@@ -80,6 +82,23 @@
...
+		if (!isturf(T))
+			return
+		if(last_act + 20 > world.time)//prevents message spam
+			return
+		last_act = world.time
+		to_chat(user, "<span class='notice'>You start pulling out pieces of rock with your hands...</span>")
+
+		if(do_after(user, 15 SECONDS, target = src))
+			if(ismineralturf(src))
+				to_chat(user, "<span class='notice'>You finish pulling apart the rock.</span>")
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:97]:
```suggestion
				to_chat(user, "<span class='notice'>You finish pulling apart [src].</span>")
```

File: code/game/turfs/closed/minerals.dm [hunk 7]
```diff
@@ -80,6 +82,23 @@
...
 
+/turf/closed/mineral/attack_hand(mob/user)
+	if(weak_turf)
+		var/turf/T = user.loc
+		if (!isturf(T))
+			return
+		if(last_act + 20 > world.time)//prevents message spam
+			return
+		last_act = world.time
+		to_chat(user, "<span class='notice'>You start pulling out pieces of rock with your hands...</span>")
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:93]:
```suggestion
		to_chat(user, "<span class='notice'>You start pulling out pieces of [src] with your hands...</span>")
```

File: code/game/turfs/open/openspace.dm [hunk 8]
```diff
@@ -152,17 +152,25 @@ GLOBAL_DATUM_INIT(openspace_backdrop_one_for_all, /atom/movable/openspace_backdr
 	initial_gas_mix = ICEMOON_DEFAULT_ATMOS
 	planetary_atmos = TRUE
 	var/replacement_turf = /turf/open/floor/plating/asteroid/snow/icemoon
+	var/protect_ruin = TRUE
+	var/drill_below = TRUE
```
Comment (Mothblocks) [inline code/game/turfs/open/openspace.dm:156]: Document.

File: code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm [hunk 9]
```diff
@@ -51,6 +54,8 @@ Difficulty: Extremely Hard
 
 /mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/Initialize()
 	. = ..()
+	for(var/obj/structure/frost_miner_prism/P in GLOB.frost_miner_prisms)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:57]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm [hunk 10]
```diff
@@ -77,7 +82,7 @@ Difficulty: Extremely Hard
 
 /mob/living/simple_animal/hostile/megafauna/demonic_frost_miner/OpenFire()
 	check_enraged()
-	projectile_speed_multiplier = 1 - enraged * 0.25
+	projectile_speed_multiplier = 1 - enraged * 0.5 // idk why i nerfed this in testing but its better this way
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:85]:
```suggestion
	projectile_speed_multiplier = 1 - enraged * 0.5
```

Comments don't need to talk about what code used to be.

File: code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm [hunk 11]
```diff
@@ -252,6 +257,8 @@ Difficulty: Extremely Hard
 	icon_state = "demonic_miner_phase2"
 	animate(src, pixel_y = pixel_y - 96, time = 8, flags = ANIMATION_END_NOW)
 	spin(8, 2)
+	for(var/obj/structure/frost_miner_prism/P in GLOB.frost_miner_prisms)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:260]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm [hunk 12]
```diff
@@ -265,6 +272,8 @@ Difficulty: Extremely Hard
 		return
 	var/turf/T = get_turf(src)
 	var/loot = rand(1, 3)
+	for(var/obj/structure/frost_miner_prism/P in GLOB.frost_miner_prisms)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:275]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm [hunk 13]
```diff
@@ -376,12 +385,17 @@ Difficulty: Extremely Hard
...
+	duration = 4 SECONDS
+	status_type = STATUS_EFFECT_REPLACE
 	alert_type = /atom/movable/screen/alert/status_effect/ice_block_talisman
 	/// Stored icon overlay for the hit mob, removed when effect is removed
 	var/icon/cube
 
+/datum/status_effect/ice_block_talisman/on_creation(mob/living/new_owner, set_duration)
+	if(isnum(set_duration))
+		duration = set_duration
+	. = ..()
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/megafauna/demonic_frost_miner.dm:397]:
```suggestion
	return ..()
```

File: code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm [hunk 14]
```diff
@@ -276,19 +276,26 @@
...
 
 //fire line keeps going even if dragon is deleted
-/proc/dragon_fire_line(atom/source, list/turfs)
+/proc/dragon_fire_line(atom/source, list/turfs, frozen = FALSE)
 	var/list/hit_list = list()
 	for(var/turf/T in turfs)
 		if(istype(T, /turf/closed))
 			break
-		new /obj/effect/hotspot(T)
+		var/obj/effect/hotspot/H = new /obj/effect/hotspot(T)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm:284]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/ice demon.dm [hunk 15]
```diff
@@ -46,20 +46,22 @@
...
 		return ..()
 	if(teleport_distance <= 0)
 		return ..()
-	var/list/possible_ends = list()
-	for(var/turf/T in view(teleport_distance, target.loc) - view(teleport_distance - 1, target.loc))
-		if(isclosedturf(T))
+	var/list/possible_ends = view(teleport_distance, target.loc) - view(teleport_distance - 1, target.loc)
+	for(var/turf/T in possible_ends)
+		if(isopenturf(T))
 			continue
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/ice demon.dm:63]:
```suggestion
	for(var/turf/closed/T in possible_ends)
```

Then, https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/ice whelp.dm [hunk 16]
```diff
@@ -39,10 +40,10 @@
 	/// How far the whelps fire can go
 	var/fire_range = 4
 
-/mob/living/simple_animal/hostile/asteroid/ice_whelp/OpenFire()
+/mob/living/simple_animal/hostile/asteroid/ice_whelp/Shoot()
 	var/turf/T = get_ranged_target_turf_direct(src, target, fire_range)
 	var/list/burn_turfs = getline(src, T) - get_turf(src)
-	dragon_fire_line(src, burn_turfs)
+	dragon_fire_line(src, burn_turfs, TRUE)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/ice whelp.dm:46]: Named variable for `TRUE`.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Whoneedspacee** [comment]
> thats what you get for not atomizing ur stuff

L

File: code/game/turfs/closed/minerals.dm [hunk 17]
```diff
@@ -1,3 +1,5 @@
+#define MINING_MESSAGE_COOLDOWN 20
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:1]: Undef at the end.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**